application/frontend/controllers/AuthController.php

```diff
@@ -63,7 +64,12 @@ public function actionLogin()
             try {
                 $clientId = Utils::getClientIdOrFail();
             } catch (\Exception $e) {
-                throw new BadRequestHttpException(\Yii::t('app', 'Auth.MissingClientID'), 1545316879);
+                \Yii::warning(\Yii::t('app', 'Auth.MissingClientID'));
+
+                // This condition happens if a user sits on the IDP login prompt long
+                // enough for the session to expire. As a workaround, redirect back to
+                // a profile UI page, which should restart the login process
+                return $this->redirect($afterLogin);
```

Comment:
I believe these changes will work, but there are enough possible paths through this code that I'm not sure I've considered them all.

Since the scenario we're trying to improve here is someone leaving their browser on the login page for over 30 minutes and then trying to log in, I'm wondering about ignoring the Return-To URL and simply sending them just to the plain UI_URL. Thoughts?

I think the question comes down to whether we trust the Return-To URL more/less depending on whether the user has/hasn't already authenticated. We already (currently) ensure that it begins with the UI_URL, so I don't know whether it matters... just trying to think through the repercussions of the change.

Reply:
I like that suggestion. Thanks.